sklearn/manifold/tests/test_t_sne.py

```diff
@@ -263,3 +389,90 @@ def test_chebyshev_metric():
+    random_state = check_random_state(0)
+    X = random_state.randn(5, 2)
+    for nc in [4, 100]:
+        tsne = TSNE(n_iter=199, method='barnes_hut', n_components=nc)
+        m = ".*method='barnes_hut' only available for.*"
+        assert_raises_regexp(ValueError, m, tsne.fit_transform, X)
+
+
+def test_64bit():
+    """Test to ensure 64bit arrays are handled correctly"""
+    random_state = check_random_state(0)
+    methods = ['barnes_hut', 'standard']
+    for method in methods:
+        for dt in [np.float32, np.float64]:
+            X = random_state.randn(100, 2).astype(dt)
+            tsne = TSNE(n_components=2, perplexity=2, learning_rate=100.0,
+                        random_state=0, method=method)
+            tsne.fit_transform(X)
+
+
+def test_transform_before_fit():
+    """transform() cannot be called before fit()"""
+    random_state = check_random_state(0)
+    X = random_state.randn(100, 2)
+    tsne = TSNE(n_components=2, perplexity=2, learning_rate=100.0,
+                random_state=0, method='barnes_hut')
+    m = ".*Cannot call `transform` unless `fit` has.*"
+    assert_raises_regexp(ValueError, m, tsne.transform, X)
+
+
+def test_transform_warning():
+    """Raises a warning if fit and transform encountered the same data """
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        random_state = check_random_state(0)
+        X = random_state.randn(100, 2)
+        tsne = TSNE(n_components=2, perplexity=2, learning_rate=100.0,
+                    random_state=0, method='barnes_hut')
+        tsne.fit(X)
+        tsne.transform(X)
+        m = str(w[0].message)
+    assert "The transform input appears to be similar" in m
+
+
+def test_quadtree_similar_point():
+    """
+    Test that a point can be introduced into a quad tree
+    where a similar point already exists.
+
+    Test will hang if it doesn't complete.
+    """
+
+    Xs = []
+    # check the case where points are actually different
+    Xs.append(np.array([[1, 2], [3, 4]]))
+    # check the case where points are the same on X axis
+    Xs.append(np.array([[-9.368728, 10.264389], [-9.368728, 11.264389]]))
+    # check the case where points are arbitraryily close on X axis
+    Xs.append(np.array([[-9.368728, 10.264389], [-9.368761, 11.264389]]))
+    # check the case where points are the same on Y axis
+    Xs.append(np.array([[-10.368728, 3.264389], [-11.368761, 3.264389]]))
+    # check the case where points are arbitrarily close on Y axis
+    Xs.append(np.array([[-10.368728, 3.264339], [-11.368761, 3.264389]]))
+    # check the case where points are arbitraryily close on both axes
+    Xs.append(np.array([[-9.368728, 3.264389], [-9.368761, 3.264389]]))
+
+    for X in Xs:
+        counts = np.zeros(3, dtype='int64')
+        _barnes_hut_tsne.check_quadtree(X, counts)
+        m = "Tree consistency failed: unexpected number of points at root node"
+        assert counts[0] == counts[1], m
```

Comment:
Please use `assert_equal` to get informative failure message with the expected and observed count values.